From c1e63055bdd0568698086c3a3b2d493d27673fea Mon Sep 17 00:00:00 2001
From: Thierry Vignaud <thierry.vignaud@gmail.com>
Date: Wed, 11 May 2016 22:27:52 +0200
Subject: [PATCH] fix segfault when calling with args==NULL

the doc explicitely describe "args" as "signing parameters (or NULL for
defaults)"

This no more true since commit 6e9eab345a18d01f0c2c35fa8bb842744093364c
As such, with rpm-4.13, some callers will segfault (eg: perl-RPM4's
testsuite)
---
 sign/rpmgensig.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/sign/rpmgensig.c b/sign/rpmgensig.c
index 9341fa8..2e15530 100644
--- a/sign/rpmgensig.c
+++ b/sign/rpmgensig.c
@@ -870,7 +870,7 @@ int rpmPkgSign(const char *path, const struct rpmSignArgs * args)
 	}
     }
 
-    rc = rpmSign(path, 0, args->signfiles);
+    rc = rpmSign(path, 0, args ? args->signfiles : 0);
 
     if (args) {
 	if (args->hashalgo) {
-- 
2.8.2

